armi/bookkeeping/db/__init__.py

```diff
@@ -67,6 +67,15 @@
 from .factory import databaseFactory
 
 
+__all__ = [
```

Comment:
Might be good to add a comment describing why you need to redefine what the import * does for this module.

Reply:
It doesn't redefine, it /defines/ what import * does for the package. So adding a comment to that effect would be duplicating the python docs for `__all__`

Reply:
Sure, I understand that, but there is some default behavior. `from armi.bookkeping.db import *` will do something whether or not you define __all__. This is why I used the word "redefine", but I don't want to split hairs on semantics.  

I wasn't saying you should explain what this does. That is obvious, but rather **why** you are doing it.  Why do we need special behavior here?